### PR TITLE
AB#2998 - intake communication preference form only

### PR DESCRIPTION
**Pull request by leiye29**
### Description
communication preference form skeleton 

### Screenshots (if applicable)

image
image

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  };
+
+  const nonDigitalMethodHandler = () => {
+    setDigitalMethodChecked(false);
+  };
+
+  return (
+    <>
+      <p>{t('communication-preference:note')}</p>
+      <Form method="post">
```
Comment (gregory-j-baker) [inline frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx:85]: Add a `noValidate` flag to the `<Form>` component here.
[see @sebastien-comeau's PR](https://github.com/DTS-STN/canadian-dental-care-plan/pull/529)

**Review by gregory-j-baker — DISMISSED** [review]
Just one tiny change requested. 🥇

**Comment by gregory-j-baker** [comment]
> Just one tiny change requested. 🥇

Actually just one other comment, and it's a bit of a nitpick.. I feel like the top margin of the email address inputs should be increased.

From
image

To something like
image

I used `my-4` in that screenshot (replaced `mb-2`). What do you think?

File: frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    { labelI18nKey: 'communication-preference:breadcrumbs.preferred-communication' },
+  ],
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-00XX',
+  pageTitleI18nKey: 'communication-preference:page-title',
+} as const satisfies RouteHandleData;
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
```
Comment (nicholas-ly) [inline frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx:35]: We can remove any references to RAOIDC for the public intake form as there is no authentication through ECAS. This same comment applies to the `action(..)` function.

File: frontend/public/locales/en/communication-preference.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "page-title": "Communication preferences",
+  "breadcrumbs": {
+    "preferred-communication": "Communication preferences"
+  },
+  "id": "ID",
```
Comment (nicholas-ly) [inline frontend/public/locales/en/communication-preference.json:6]: Did you intend on using this i18n key anywhere? It's currently not being referenced. 

(Same comment for French)

File: frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        <div className="my-6">
+          {preferredCommunicationMethods.length > 0 && (
+            <fieldset id="input-radios=preferred-method" className="mb-6" data-testid="input-radios-preferred-method">
+              <InputLegend id="input-radios=preferred-method-legend" className="mb-2" required>
+                {t('communication-preference:preferred-method')}
+              </InputLegend>
+              <ul className="space-y-2">
+                <li>
+                  <InputRadio id="preferred-method-0" name="preferredMethod" value="digital" onClick={digitalMethodHandler}>
+                    {getNameByLanguage(i18n.language, preferredCommunicationMethods[0])}
```
Comment (sebastien-comeau) [inline frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx:95]: You may want to utilize array.find to retrieve the corresponding label for the preferred communication method. Look up by a stable identifier, such as a code, to ensure consistency. If the order of REST calls changes, relying on the order may lead to incorrect labels. Considering that the rest of the form section is hardcoded, using a lookup table for just one element raises concerns. It's important to maintain consistency: either base everything on the lookup table or keep everything hardcoded.

File: frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx (hunk 2 above)
Comment (leiye29) [reply frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx:32]: removed

File: frontend/public/locales/en/communication-preference.json (hunk 3 above)
Comment (leiye29) [reply frontend/public/locales/en/communication-preference.json:3]: removed

File: frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx (hunk 4 above)
Comment (leiye29) [reply frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx]: updated

File: frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx (hunk 1 above)
Comment (leiye29) [reply frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx]: added

**Review by nicholas-ly — APPROVED** [review]
FYI we can now use the `append` prop on an `<InputRadio />` to add extra content such as the input fields we have in the "Preferred Method of Communication" section.

See https://github.com/DTS-STN/canadian-dental-care-plan/pull/543

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**